middlewares/osio/auth.go

```diff
@@ -49,8 +58,8 @@ type SecretLocator interface {
 type TokenTypeLocator func(string) (TokenType, error)
 
 type cacheData struct {
-	Token    string
-	Location string
+	Token     string
+	Namespace namespace
```

Comment:
Now instead of only Location whole Namespace object is cached which contains api_url, metrics_url, console_url.  So once namespace is cached for particular user, all urls will be available without making http call.